src/providers/haskell.rs

```diff
@@ -0,0 +1,76 @@
+use std::collections::BTreeMap;
+
+use serde::Deserialize;
+
+use crate::nixpacks::{
+    nix::Pkg,
+    phase::{BuildPhase, InstallPhase, SetupPhase, StartPhase},
+};
+
+use super::Provider;
+
+pub struct HaskellStackProvider {}
+
+impl Provider for HaskellStackProvider {
+    fn name(&self) -> &str {
+        "haskell_stack"
+    }
+
+    fn detect(
+        &self,
+        app: &crate::nixpacks::app::App,
+        _env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<bool> {
+        Ok(app.includes_file("package.yaml"))
+    }
+
+    fn setup(
+        &self,
+        _app: &crate::nixpacks::app::App,
+        _env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<Option<crate::nixpacks::phase::SetupPhase>> {
+        Ok(Some(SetupPhase::new(vec![
+            Pkg::new("stack"),
+            Pkg::new("apt"),
```

Comment:
Why is `apt` needed?

Reply:
`apt` is needed in case we’re using a base image other than Debian at some point in the future.